src/services/refactors/convertArrowFunctionOrFunctionExpression.ts

```diff
@@ -0,0 +1,185 @@
+            actions: possibleActions
+        }];
+    }
+
+    function getEditsForAction(context: RefactorContext, actionName: string): RefactorEditInfo | undefined {
+        const { file, startPosition } = context;
+        const info = getInfo(file, startPosition);
+
+        if (!info) return undefined;
+        const { func } = info;
+
+        let body: Block | ConciseBody;
+        let newNode: ArrowFunction | FunctionExpression | FunctionDeclaration;
+        let edits: FileTextChanges[] = [];
+
+        switch (actionName) {
+            case toAnonymousFunctionActionName:
+
+                body = makeBlock(func);
+                newNode = createFunctionExpression(func.modifiers, func.asteriskToken, /* name */ undefined, func.typeParameters, func.parameters, func.type, body);
+                edits = textChanges.ChangeTracker.with(context, t => t.replaceNode(file, func, newNode));
+                break;
+
+            case toNamedFunctionActionName:
+
+                body = makeBlock(func);
+                const headInfo = getHeaderInfo(func);
+                if (!headInfo) return undefined;
+
+                const { variableDeclaration, variableDeclarationList, statement, name } = headInfo;
+                newNode = makeFunctionDeclaration(func, statement, name, body);
+
+                if (variableDeclarationList.declarations.length === 1) {
+                    edits = textChanges.ChangeTracker.with(context, t => t.replaceNode(file, statement, newNode));
+                }
+                else {
+                    edits = textChanges.ChangeTracker.with(context, t => {
+                        t.delete(file, variableDeclaration);
+                        t.insertNodeAfter(file, statement, newNode);
+                    });
+                }
+                break;
+
+            case toArrowFunctionActionName:
+                if (!isFunctionExpression(func)) return undefined;
+
+                const statements = func.body.statements;
+                const head = statements[0];
+                if (func.body.statements.length === 1 && (isReturnStatement(head) || isExpressionStatement(head))) {
+                    body = head.expression!;
+                    suppressLeadingAndTrailingTrivia(body);
+                    copyComments(head, body, file, SyntaxKind.MultiLineCommentTrivia, /* hasTrailingNewLine */ false);
+                }
+                else {
+                    body = func.body;
+                }
+
+                newNode = createArrowFunction(func.modifiers, func.typeParameters, func.parameters, func.type, /* equalsGreaterThanToken */ undefined, body);
+                edits = textChanges.ChangeTracker.with(context, t => t.replaceNode(file, func, newNode));
+                break;
+
+            default:
+                Debug.fail("invalid action");
+                break;
+        }
+
+        return { renameFilename: undefined, renameLocation: undefined, edits };

```

Comment:
```suggestion
        return undefined;
```
Reasoning:
- other cases returned above